apps/docs/pages/guides/getting-started/quickstarts/kotlin.mdx

```diff
@@ -0,0 +1,298 @@
+        }
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={5}>
+
+    <StepHikeCompact.Details title="Initialize the Supabase client">
+    You can create a whenever you need to perform some API call.
+    However, it is recommended that we should provide this kind of instance by using dependency injection library like Hilt. So see the code below to provide them properly. This is better for testability and maintainability
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+
+      ```kotlin
+      // Handy create instance when need
+      val client = createSupabaseClient(
+          supabaseUrl = "https://xyzcompany.supabase.co", 
+          supabaseKey = "public-anon-key"
+        ) {
+              install(GoTrue)
+              install(Postgrest)
+              install(Storage)
+          }
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={6}>
+
+    <StepHikeCompact.Details title="Create data transfer object">
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+      @Serializable
+      data class ProductDto(
+          @SerialName("productid")
+          val productId: String,
+          @SerialName("name")
+          val name: String,
+          @SerialName("description")
+          val description: String,
+          @SerialName("price")
+          val price: Double,
+          @SerialName("image")
+          val image: String,
+          @SerialName("category")
+          val category: String,
+          @SerialName("nutrition")
+          val nutrition: String,
+          @SerialName("_id")
+          val _id: Int,
+      )
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={7}>
+
+    <StepHikeCompact.Details title="Create domain object">
+      This kind of object will be consumed by the view
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+      data class Product(
+          val productId: String,
+          val name: String,
+          val description: String,
+          val price: Double,
+          val image: String,
+          val category: String,
+          val nutrition: String,
+          val _id: Int,
+      )
+      ```
+
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={8}>
+
+    <StepHikeCompact.Details title="Query data from the app">
+      
+    </StepHikeCompact.Details>
+
+    <StepHikeCompact.Code>
+      ```kotlin
+      suspend fun getProducts(): ProductDto {
+        val result = client.postgrest["products"]
+            .select().decodeList<ProductDto>()
+        // Handle result data for next step
+        return result
+      }
+      ```
+    </StepHikeCompact.Code>
+
+  </StepHikeCompact.Step>
+
+  <StepHikeCompact.Step step={9}>
+    <StepHikeCompact.Details title="Get data from ViewModel inside a coroutine scope">
+     
```

Comment:
As you are using Hilt, you may need a description saying that you are using Hilt in this case for dependency injection. See *link* to learn more about Hilt